### ISSUE TITLE
[CI] Multiple test failures JDK22 on 8.12

### ISSUE THREAD

**Issue by benwtrent** [elastic/elasticsearch#106022]
Multiple test failures due to "Unsupported class file major version 66".

**Build scan:**
https://gradle-enterprise.elastic.co/s/hkm5o5zei6wts/tests/:libs:elasticsearch-plugin-scanner:test/org.elasticsearch.plugin.scanner.ClassReadersTests/testMultipleJarsInADir

**Reproduction line:**
```
./gradlew ':libs:elasticsearch-plugin-scanner:test' --tests "org.elasticsearch.plugin.scanner.ClassReadersTests.testMultipleJarsInADir" -Dtests.seed=5E261D9284F491C2 -Dtests.locale=hu-HU -Dtests.timezone=America/Argentina/Catamarca -Druntime.java=22
```

**Applicable branches:**
8.12

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.plugin.scanner.ClassReadersTests#testMultipleJarsInADir`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testMultipleJarsInADir'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.plugin.scanner.ClassReadersTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure excerpt:**
```
java.lang.IllegalArgumentException: Unsupported class file major version 66

  at __randomizedtesting.SeedInfo.seed([5E261D9284F491C2:8A296B192E6D6168]:0)
  at org.objectweb.asm.ClassReader.<init>(ClassReader.java:199)
  at org.objectweb.asm.ClassReader.<init>(ClassReader.java:180)
  at org.objectweb.asm.ClassReader.<init>(ClassReader.java:166)
  at org.elasticsearch.plugin.scanner.ClassReaders.lambda$classesInPath$5(ClassReaders.java:97)
  at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:212)
  at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:194)
  at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:194)
  at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:194)
  at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:212)
  at java.util.Iterator.forEachRemaining(Iterator.java:133)
  at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1939)
  at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:556)
  at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:546)
  at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
  at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
  at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:702)
  at org.elasticsearch.plugin.scanner.ClassReaders.classesInPath(ClassReaders.java:102)
  at org.elasticsearch.plugin.scanner.ClassReaders.classesInJar(ClassReaders.java:83)
  at org.elasticsearch.plugin.scanner.ClassReaders.lambda$ofPaths$1(ClassReaders.java:73)
  at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:288)
  at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:194)
  at java.util.Iterator.forEachRemaining(Iterator.java:133)
  at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1939)
  at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:556)
  at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:546)
  at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:622)
  at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:291)
  at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:631)
  at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:637)
  at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:642)
  at org.elasticsearch.plugin.scanner.ClassReaders.ofPaths(ClassReaders.java:77)
  at org.elasticsearch.plugin.scanner.ClassReaders.ofDirWithJars(ClassReaders.java:48)
  at org.elasticsearch.plugin.scanner.ClassReadersTests.testMultipleJarsInADir(ClassReadersTests.java:125)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1570)

```

**Comment by rjernst** [comment]
This does not reproduce for me. I'm passing back to delivery since it is clearly build related.

**Comment by rjernst** [comment]
This was already fixed for 8.13 earlier in https://github.com/elastic/elasticsearch/pull/104085. I've opened https://github.com/elastic/elasticsearch/pull/106033 to backport that change.

**Comment by mark-vieira** [comment]
Fixed in 8.13 and 8.12 branch is EOL.
